src/FormCheckInput.tsx

```diff
@@ -63,10 +73,26 @@ const FormCheckInput: BsPrefixRefForwardingComponent<
     const { controlId } = useContext(FormContext);
     bsPrefix = useBootstrapPrefix(bsPrefix, 'form-check-input');
 
+    const inputRef = React.useRef<HTMLInputElement>();
+    const existingRef =
```

Comment:
I think it's better to use `useMergedRefs` here?   Also do we need to check with `isForwardRef` and `isDOMElement`?  Maybe we leave this up to the user to ensure `Component` is a proper ref forwarded component

Reply:
I think the issue is we always need to attach a ref, or maybe it's enough to check if `indeterminate` is provided and then leave it up to the user to make it reffable?

Reply:
Yeah that sounds good.  We can avoid the new dep

Reply:
Just to be sure, there are 3 cases i am trying to cover here
- `as` prop is a html tag - no problem, using `ref || existingRef` will work(`existingRef` will be passed)
- `as` prop is a React element which accepts a `ref` - no problem, using ` ref || existingRef` will work(`ref` will be passed)
- `as` props is a React element which does not accept a `ref` - we will try to pass a ref(`existingRef`) to an element which doesn't accept a `ref` - this is the only edge case where I'd pass `ref={undefined}`

I can leave the 3rd case broken and require the user to provide a react component that forwards a ref, but we'd lose on backwards-compatibility front(i even had to remove one test to make this work).

Reply:
by checking something along of what i added which can be simplified to: `isForwardRef(Component) ? undefined : someRef` we can support every case i think, the downside being the `react-is` package

Reply:
I was suggesting only adding the ref if `indeterminate != undefined` that should be backwards compatible since the prop is new. you can do something like:

```
const inputRef = useRef(null)
const mergedRef = useMergedRef(inputRef, outerRef)

<Component ref={indeterminate != null || outerRef ? mergedRef : undefined}
```

check tho that `ref={undefined}` doesn't still send the error sometimes React uses `in` checks isntead